nomos-core/chain-defs/src/header/leader_proof.rs

```diff
@@ -0,0 +1,180 @@
+// std
+// crates
+use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+use thiserror::Error;
+// internal
+use nomos_ledger::leader_proof::LeaderProof;
+use nomos_proof_statements::leadership::{LeaderPrivate, LeaderPublic};
+
+#[derive(Debug, Clone)]
+pub struct Risc0LeaderProof {
+    public_inputs: LeaderPublic,
+    risc0_receipt: risc0_zkvm::Receipt,
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Risc0 proof failed: {0}")]
+    Risc0ProofFailed(#[from] anyhow::Error),
+}
+
+impl Risc0LeaderProof {
+    pub fn prove(
+        public_inputs: LeaderPublic,
+        private_inputs: LeaderPrivate,
+        prover: &dyn risc0_zkvm::Prover,
+    ) -> Result<Self, Error> {
+        let env = risc0_zkvm::ExecutorEnv::builder()
+            .write(&public_inputs)
+            .unwrap()
+            .write(&private_inputs)
+            .unwrap()
+            .build()
+            .unwrap();
+
+        let start_t = std::time::Instant::now();
+
+        // ATTENTION: producing a groth16 proof currently requires x86 with docker support
```

Comment:
They use virtualization for this? I'm really curious why is the case. 🤔

Reply:
Not sure why, I guess it's just to bundle dependencies for rapidsnark which is what they run inside the container